block/deserialize.go

```diff
@@ -0,0 +1,341 @@
+package block
+
+import (
+	"bytes"
+	"errors"
+
+	"github.com/vulpemventures/go-elements/internal/bufferutil"
+
+	"github.com/vulpemventures/go-elements/transaction"
+)
+
+func deserialize(buf *bytes.Buffer) (*Block, error) {
+	header, err := DeserializeHeader(buf)
+	if err != nil {
+		return nil, err
+	}
+
+	transactions, err := DeserializeTransactions(buf)
+	if err != nil {
+		return nil, err
+	}
+
+	return &Block{
+		Header: header,
+		TransactionsData: &Transactions{
+			Transactions: transactions,
+		},
+	}, nil
+}
+
+func DeserializeHeader(
+	buf *bytes.Buffer,
+) (*Header, error) {
+	d := bufferutil.NewDeserializer(buf)
+
+	version, err := d.ReadUint32()
+	if err != nil {
+		return nil, err
+	}
+
+	isDyna := false
+	if version>>31 == 1 {
+		isDyna = true
+	}
```

Comment:
this could become: 
```suggestion
	isDyna := version>>31 == 1
```
couldn't it?

Reply:
Done